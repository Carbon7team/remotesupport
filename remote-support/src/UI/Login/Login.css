.login-wrapper {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: #fafafa;
}

.login-wrapper label,
.login-wrapper legend{
    font-size: 1em;
    font-weight: bold;
    padding: 0.4em;
    color: #333;
}