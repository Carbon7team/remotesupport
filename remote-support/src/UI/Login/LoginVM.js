class LoginVM{

    constructor(rootstore, username, password){
        this.rootstore = rootstore;
        this.username = username;
        this.password = password;
    }

    async loginFetch(username , password) {
      var credentials = {username, password}
        return fetch('http://localhost:8080/login', {
          method: 'POST',
          headers: { 'Content-Type': 'application/json'},
          body: JSON.stringify(credentials)
        })
          .then(data => data.json())
      }

    handleSubmit = async e => {
        e.preventDefault();
        // eslint-disable-next-line
        const token = await loginFetch(this.username, this.password);
      }
}

export default LoginVM;