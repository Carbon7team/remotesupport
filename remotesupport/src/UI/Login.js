import React from 'react';
import { useState } from 'react';
import PropTypes from 'prop-types';

async function loginFetch(credentials) {
  return fetch('http://localhost:8080/login', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json'},
    body: JSON.stringify(credentials)
  })
    .then(data => data.json())
}

export default function Login({setToken}) {
  const [username, setUsername] = useState();
  const [password, setPassword] = useState();

  const handleSubmit = async e => {
    e.preventDefault();
    const token = await loginFetch({username, password});
    setToken(token);
  }

  return(
    <div className="login-wrapper">

      <h1>Please Log In</h1>

      <form onSubmit={handleSubmit}>

      <fieldset>

      <legend>Personal data</legend>

      <div className="username-wrapper">
        <label htmlFor="username">Username</label>
        <span><input type="text" name="username" id="username" placeholder="name.surname" onChange={e => setUsername(e.target.value)}/></span>
      </div>

      <div className="username-wrapper">
        <label htmlFor="password">Password</label>
        <span><input type="password" name="password" id="password" onChange={e => setPassword(e.target.value)}/></span>
      </div>

      <input type="submit" name="submit" value="Sign in" />

      </fieldset>

      </form>

    </div>
  )
}

Login.propTypes = { setToken: PropTypes.func.isRequired }