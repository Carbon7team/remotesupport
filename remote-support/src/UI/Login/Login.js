import { observer } from "mobx-react-lite";
import React from "react";
import { useState } from "react";
import { useStore } from "../../Utilities/contextProvider";
import "./Login.css";

const Login = observer(() => {
  const rootstore = useStore();

  const [username, setUsername] = useState("");

  const [password, setPassword] = useState("");

  const [errorMessage, setErrorMessage] = useState("");

  async function loginFetch(username, password) {
    var credentials = { username, password };
    return fetch("http://localhost:4000", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(credentials),
    }).then((data) => data.json());
  }

  const handleSubmit = async (e) => {
    e.preventDefault();
    const response = JSON.parse(await loginFetch(username, password));
    if (response.token != null) {
      rootstore.stateUIStore.setLogged(true);
      rootstore.stateUIStore.setIdTech(response.user);
      rootstore.stateUIStore.setTokenAuth(response.token);
      setErrorMessage(response.message);
    }
  };

  return (
    <div className="login-wrapper">
      <h1>Welcome technician,</h1>

      <p>Sign in</p>

      <form onSubmit={handleSubmit}>
        <fieldset>
          <legend>Personal data</legend>

          <div className="username-wrapper">
            <label htmlFor="username">Username</label>
            <span>
              <input
                type="text"
                name="username"
                id="username"
                placeholder="name.surname"
                onChange={(e) => setUsername(e.target.value)}
              />
            </span>
          </div>

          <div className="username-wrapper">
            <label htmlFor="password">Password</label>
            <span>
              <input
                type="password"
                name="password"
                id="password"
                onChange={(e) => setPassword(e.target.value)}
              />
            </span>
          </div>

          <input type="submit" name="submit" value="Sign in" />

          <div id="error-message">
            <p>{errorMessage}</p>
          </div>
        </fieldset>
      </form>
    </div>
  );
});

export default Login;
