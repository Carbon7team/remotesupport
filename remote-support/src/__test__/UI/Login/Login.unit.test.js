import { create } from "react-test-renderer";
import Login from "../../../UI/Login/Login";

describe("Login render", () => {
  test("Login render", () => {
    let tree = create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
