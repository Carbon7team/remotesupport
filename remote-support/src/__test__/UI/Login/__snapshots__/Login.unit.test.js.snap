// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render Login render 1`] = `
<div
  className="login-wrapper"
>
  <h1>
    Welcome technician,
  </h1>
  <p>
    Sign in
  </p>
  <form
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        Personal data
      </legend>
      <div
        className="username-wrapper"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <span>
          <input
            id="username"
            name="username"
            onChange={[Function]}
            placeholder="name.surname"
            type="text"
          />
        </span>
      </div>
      <div
        className="username-wrapper"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <span>
          <input
            id="password"
            name="password"
            onChange={[Function]}
            type="password"
          />
        </span>
      </div>
      <input
        name="submit"
        type="submit"
        value="Sign in"
      />
      <div
        id="error-message"
      >
        <p>
          
        </p>
      </div>
    </fieldset>
  </form>
</div>
`;
